source/coll_dist.f90

```diff
@@ -0,0 +1,443 @@
+! ================================================================================================ !
+!  Dist Module for Collimation
+!  Extracted from collimation module
+! ================================================================================================ !
+module coll_dist
+
+  use parpro
+  use floatPrecision
+  use numerical_constants
+
+  implicit none
+
+  ! Other Settings
+  integer,                  public,  save :: cdist_logUnit  = -1
+  real(kind=fPrec),         public,  save :: cdist_energy   = zero
+
+  ! Twiss
+  real(kind=fPrec),         public,  save :: cdist_alphaX   = zero
+  real(kind=fPrec),         public,  save :: cdist_alphaY   = zero
+  real(kind=fPrec),         public,  save :: cdist_betaX    = zero
+  real(kind=fPrec),         public,  save :: cdist_betaY    = zero
+  real(kind=fPrec),         public,  save :: cdist_emitX    = zero
+  real(kind=fPrec),         public,  save :: cdist_emitY    = zero
+
+  ! Distribution Settings
+  real(kind=fPrec),         public,  save :: cdist_ampX     = zero
+  real(kind=fPrec),         public,  save :: cdist_ampY     = zero
+  real(kind=fPrec),         public,  save :: cdist_ampR     = zero
+  real(kind=fPrec),         public,  save :: cdist_smearX   = zero
+  real(kind=fPrec),         public,  save :: cdist_smearY   = zero
+  real(kind=fPrec),         public,  save :: cdist_smearR   = zero
+  real(kind=fPrec),         public,  save :: cdist_spreadE  = zero
+  real(kind=fPrec),         public,  save :: cdist_bunchLen = zero
+
+  character(len=mFileName), public,  save :: cdist_fileName = " "
+
+contains
+
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-03-26
+!  Updated: 2019-04-16
+!  Generate the distribution in the correct order, based on the distFormat flag
+! ================================================================================================ !
+subroutine cdist_makeDist(distFormat)
+
+  use crcoall
+
+  integer, intent(in) :: distFormat
+
+  select case(distFormat)
+  case(0)
+    return
+  case(1)
+    call cdist_makeDist_fmt1
+  case(2)
+    call cdist_makeDist_fmt1
+    call cdist_makeDist_fmt2
+  case(3)
+    call cdist_makeDist_fmt1
+    call cdist_makeDist_fmt2
+    call cdist_makeDist_fmt3
+  case(4)
+    call cdist_makeDist_fmt4
+  case(5)
+    call cdist_makeDist_fmt3
+    call cdist_makeDist_fmt5
+  case(6)
+    call cdist_makeDist_fmt4
+    call cdist_makeDist_fmt6
+  case default
```

Comment:
the naming convention could be reviewed, as (in the end) the actual format is accomplished only if you have all the calls performed regularly